lib/evaluation/limitDisclosureEvaluationHandler.ts

```diff
@@ -0,0 +1,148 @@
+    for (let i = 0; i < fields.length; i++) {
+      const field: Field = fields[i];
+      const result = JsonPathUtils.extractInputField(vc, field.path);
+      if (result.length > 0) {
+        this.copyResultPathToDestinationCredential(result[0].path, vc, vcToSend, idIdx, vcIdx);
+      } else {
+        this.createMandatoryFieldNotFoundResult(idIdx, vcIdx, field.path);
+      }
+    }
+  }
+
+  private copyResultPathToDestinationCredential(
+    pathDetails: any[],
+    verifiableCredential: unknown,
+    verifiableCredentialToSend: unknown,
+    idIdx: number,
+    vcIdx: number
+  ) {
+    let objectCursor = verifiableCredential;
+    let currentCursorInToSendObj = verifiableCredentialToSend;
+    this.createSuccessResult(idIdx, vcIdx, pathDetails);
+    for (let i = 1; i < pathDetails.length; i++) {
+      objectCursor = objectCursor[pathDetails[i]];
+      if (pathDetails.length == i + 1) {
+        currentCursorInToSendObj[pathDetails[i]] = objectCursor;
+      } else if (typeof pathDetails[i] === 'string' && typeof pathDetails[i + 1] === 'string') {
+        currentCursorInToSendObj[pathDetails[i]] = {};
+        currentCursorInToSendObj = currentCursorInToSendObj[pathDetails[i]];
+      } else if (typeof pathDetails[i] === 'string' && typeof pathDetails[i + 1] !== 'string') {
+        currentCursorInToSendObj[pathDetails[i]] = [{}];
+        currentCursorInToSendObj = currentCursorInToSendObj[pathDetails[i]];
+      } else {
+        currentCursorInToSendObj[pathDetails[i]] = {};
+        currentCursorInToSendObj = currentCursorInToSendObj[pathDetails[i]];
+      }
+    }
+  }
+
+  private copyModifiedVerifiableCredentialToExisting(verifiableCredentialToSend: any, inputDescriptorId: string) {
+    if (!this.verifiablePresentation.verifiableCredential) {
+      this.verifiablePresentation.verifiableCredential = [];
+    }
+    for (let i = 0; i < this.verifiablePresentation.presentationSubmission.descriptor_map.length; i++) {
+      const currentDescriptor: Descriptor = this.verifiablePresentation.presentationSubmission.descriptor_map[i];
+      if (currentDescriptor.id === inputDescriptorId) {
+        this.verifiablePresentation.verifiableCredential.push(verifiableCredentialToSend);
+      }
+    }
+  }
+
+  private createSuccessResult(idIdx: number, vcIdx: number, pathDetails: any[]) {
+    return this.results.push({
+      input_descriptor_path: `$.input_descriptors[${idIdx}]`,
+      verifiable_credential_path: `$.verifiableCredential[${vcIdx}]`,
+      evaluator: this.getName(),
+      status: Status.INFO,
+      message: 'added variable in the limit_disclosure to the verifiableCredential',
+      payload: pathDetails,
+    });
+  }
+
+  private createMandatoryFieldNotFoundResult(idIdx: number, vcIdx: number, path: Array<string>) {
+    return this.results.push({
+      input_descriptor_path: `$.input_descriptors[${idIdx}]`,
+      verifiable_credential_path: `$.verifiableCredential[${vcIdx}]`,
```

Comment:
Needs to be updated dynamically.

Reply:
updated after rebase

Reply:
We do not actually need to take care of path in VP wrappers at this level. For example we can deal with this while we are converting received input for evaluation by converting that to our desired format.